src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/AdapterTest/AdapterTest.cs

```diff
@@ -1255,6 +1256,52 @@ public void TestDeriveParameters()
             }
         }
 
+        [CheckConnStrSetupFact]
+        public void TestReadOnlyColumnMetadata()
+        {
+            using (SqlConnection connection = new SqlConnection(DataTestUtility.TCPConnectionString))
+            {
+                connection.Open();
+                using (SqlCommand command = connection.CreateCommand())
+                {
+                    command.CommandText = "Create table #t (ID int identity(1,1), sText varchar(12), sMemo as (convert(varchar,ID) + ' ' + sText))";
+                    command.ExecuteNonQuery();
+                }
+                using (SqlDataAdapter dataAdapter = new SqlDataAdapter("Select * from #t", connection))
+                {
+                    using (SqlCommandBuilder commandBuilder = new SqlCommandBuilder(dataAdapter))
+                    {
+                        using (SqlCommand cmd = commandBuilder.GetInsertCommand())
+                        {
+                            DataTestUtility.AssertEqualsWithDescription("INSERT INTO [#t] ([sText]) VALUES (@p1)", cmd.CommandText, "Unexpected insert command.");
```

Comment:
Why is this check needed in the test? We are essentially testing for the schema returned by the SqlDataAdapter

Reply:
It was in the test app provided by the user in issue #275 . I think it's nice to see the InsertCommand from SqlCommandBuilder gets corrected after the fix. Previously the insert command generated would've been `INSERT INTO [#t] ([sText], [sMemo]) VALUES (@p1, @p2)` since the driver didn't know sMemo was a readonly column.

Reply:
Ah I see. I completely missed this regression in the app. Thanks for pointing it out .